lib/grandpa/grandpa.go

```diff
@@ -501,25 +506,15 @@ func (s *Service) playGrandpaRound() error {
 		s.prevotes.Store(s.publicKeyBytes(), spv)
 	}
 
-	logger.Debugf("sending pre-vote message %s...", pv)
-	roundComplete := make(chan struct{})
-	// roundComplete is a signal channel which is closed when the round completes
-	// (will receive the default value of channel's type), so we don't need to
-	// explicitly send a value.
-	defer close(roundComplete)
-
-	// continue to send prevote messages until round is done
-	go s.sendVoteMessage(prevote, vm, roundComplete)
-
-	logger.Debug("receiving pre-commit messages...")
-	// through goroutine s.receiveVoteMessages(ctx)
-	time.Sleep(s.interval)
+	prevoteRoundComplete := make(chan struct{})
+	go s.sendPrevoteMessage(vm, prevoteRoundComplete)
 
 	if s.paused.Load().(bool) {
 		return ErrServicePaused
 	}
```

Comment:
what is this? this function could return early if what is returned by `s.paused.Load()` is a bool that `== true`? Shouldn't we cancel any running goroutines for this round?